patch-db/src/model.rs

```diff
@@ -9,9 +9,55 @@ use json_ptr::JsonPointer;
 use serde::{Deserialize, Serialize};
 use serde_json::Value;
 
-use crate::locker::LockType;
+use crate::{
+    bulk_locks::{self, ModelLock},
+    locker::LockType,
+    model_paths::{ModelPath, ModelPaths},
+};
 use crate::{DbHandle, DiffPatch, Error, Revision};
 
+pub trait AsPhantom<SB> {
+    fn as_phantom_binds(&self) -> Option<PhantomData<SB>>;
+}
+
+impl AsPhantom<()> for ModelPaths {
+    fn as_phantom_binds(&self) -> Option<PhantomData<()>> {
+        match self {
+            ModelPaths::Paths(_) => Some(PhantomData),
+            ModelPaths::PossibleStarPath { count, .. } if *count == 0 => Some(PhantomData),
+            _ => {
+                #[cfg(feature = "tracing")]
```

Comment:
@dr-bonez What's the pattern we want, check for an env variable to be set to throw error?